src/components/StakingItem/index.js

```diff
@@ -1 +1,246 @@
-export { default } from './StakingItem';
+import React, { useEffect, useState } from 'react';
+import { useHistory } from 'react-router';
+import PropTypes from 'prop-types';
+import { store as alertStore } from 'react-notifications-component';
+import { useWeb3React } from '@web3-react/core';
+import Collapse from '@kunukn/react-collapse';
+import { ReactSVG } from 'react-svg';
+
+import InstallMetamaskAlert from '../../pages/Home/components/InstallMetamaskAlert';
+import { formatRounded, StakingWrapper } from '../../services/staking.wrapper';
+import appStore from '../../store/app.store';
+import {
+  ethereum,
+  HIDE,
+  MAIN_PAGE,
+  SHOW,
+  STAKE,
+  STAKING_PAGE,
+} from '../../config';
+import avatarIcon from '../../assets/svg/avatar.svg';
+import Paragraph from '../Paragraph';
+import DisplayValue from '../DisplayValue';
+import Button from '../Button';
+import Deposit from '../../pages/Staking/components/Deposit/Deposit';
+import { useLogIn } from '../../hooks';
+
+const StakingItem = ({
+  expand = false,
+  activeExpand,
+  setActiveExpand,
+  state = 0,
+  dispatch,
+  index = -1,
+  poolInfo,
+}) => {
+  const { library } = useWeb3React();
+  const [myStake, setMyStake] = useState(null);
+  const [totalStake, setTotalStake] = useState(null);
+  const [APYOfPool, setAPYOfPool] = useState(null);
+  const history = useHistory();
+  const { pathname } = history.location;
+  const { active } = useWeb3React();
+  const { logIn } = useLogIn();
+  let signer;
+  const stakeBtnHandler = () => {
+    if (expand !== false) {
+      setActiveExpand(index);
+      dispatch({ type: 'toggle', index });
+      if (index === activeExpand) {
+        dispatch({ type: 'hide', index });
+      }
+      if (index === activeExpand && !state[index]) {
+        dispatch({ type: 'toggle', index });
+      }
+    } else {
+      logIn();
+    }
+  };
+
```

Comment:
Why updateState just defining const loggedInRefresh inside itself?
It's isn't called. Or I don't understand something?